lib/travis/logs/app.rb

```diff
@@ -167,11 +174,54 @@ def initialize(auth_token: ENV['AUTH_TOKEN'].to_s,
         status 204
       end
 
-      put '/logs/:job_id/archived' do
+      put '/logs/:log_id/archived' do
         halt 500, 'authentication token is not set' if auth_token.empty?
         halt 403 unless authorized?(request)
 
-        halt 501
+        data = nil
+        log_id = nil
+        archived_at = nil
+
+        begin
+          request.body.rewind
+          data = JSON.parse(request.body.read)
+          log_id = Integer(params[:log_id])
+        rescue => e
+          halt 500, JSON.dump('error' => e.to_s)
+        end
+
+        unless data.respond_to?(:key?) &&
+               data.key?('@type') &&
+               data.key?('archived_at') &&
+               data.key?('archive_verified')
+          halt 400, JSON.dump('error' => 'invalid body')
+        end
+
+        begin
+          archived_at = Time.parse(data['archived_at'])
+        rescue => e
+          halt 500, JSON.dump('error' => e.to_s)
```

Comment:
Should this be a 400-level error instead of a 500? (Same for the `JSON.parse()` and `Integer()` `begin..rescue` block a little further up.